File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.py
class Solution:
    def findLongestChain(self, pairs: List[List[int]]) -> int:
        pairs.sort(key=lambda x: x[1])
        count=1
        i=0
        j=1
        print(pairs)
        while(j<len(pairs)):
            if pairs[i][1]<pairs[j][0]:
                count+=1
                i=j
            else:
                j+=1
        return count

File: 0646-maximum-length-of-pair-chain/0646-maximum-length-of-pair-chain.java
class Solution {
    public int findLongestChain(int[][] pairs) {
        Arrays.sort(pairs, Comparator.comparingInt(a -> a[1]));
        int count = 1 , i = 0 , j = 1;
        while (j < pairs.length) {
            if (pairs[i][1] < pairs[j][0]) {
                count++;
                i = j;
            } else j++;
        }
        return count;
    }
}